### ISSUE TITLE
how to transform a estimtor to pmml if i get 1000k examples?

### ISSUE THREAD

**Issue by topxxuki** [jpmml/sklearn2pmml#105]
first , i tried SGDClassifier.partial_fit() ,it turns out that sklearn2pmml dont support partial_fit()
then i try to concatenate features ,then it just get "killed" without any infomation

**Comment by vruusmann** [comment]
> i tried SGDClassifier.partial_fit(), it turns out that sklearn2pmml dont support partial_fit()

What exactly do you mean? That the `PMMLPipeline` class should provide a `partial_fit(X, y)` method (in addition to the `fit(X, y)` method)?

You can always fit the estimator outside of the pipeline.

**Comment by topxxuki** [comment]
`    cls = SGDClassifier(loss='log',  max_iter=100, tol=0.01 ,alpha=0.01)
    pipline = PMMLPipeline([("classifier", cls)])
    for i in range(train_size):
        pipline.partial_fit(trains[2* i], trains[2*i+1], classes=([0,1]))
    print ('finish training----', time.clock())
    joblib.dump(pipline, joblib_path)
    print ('saving joblib model-----'),time.clock()
    sklearn2pmml(pipline, pmml=pmml_path)
    print ('saving pmml model-----'), time.clock()`

here is my code ,i can get the right joblib file ,while it cannot get any pmml file

**Comment by topxxuki** [comment]
`    cls = SGDClassifier(loss='log',  max_iter=100, tol=0.01 ,alpha=0.01)

    pipline = PMMLPipeline([("classifier", cls)])
    for i in range(train_size):
        pipline.partial_fit(trains[2* i], trains[2*i+1], classes=([0,1]))
    print ('finish training----', time.clock())
    joblib.dump(pipline, joblib_path)
    print ('saving joblib model-----'),time.clock()
    sklearn2pmml(pipline, pmml=pmml_path)
    print ('saving pmml model-----'), time.clock()`


here is my code ,i can get the right joblib file ,while it cannot get any pmml file
should i use cls.partial_fit(trains[2* i], trains[2*i+1], classes=([0,1])) in line 3 ?

**Comment by topxxuki** [comment]
> > i tried SGDClassifier.partial_fit(), it turns out that sklearn2pmml dont support partial_fit()
> 
> What exactly do you mean? That the `PMMLPipeline` class should provide a `partial_fit(X, y)` method (in addition to the `fit(X, y)` method)?
> 
> You can always fit the estimator outside of the pipeline.

my code are as follows:

cls = SGDClassifier(loss='log', max_iter=100, tol=0.01 ,alpha=0.01)
pipline = PMMLPipeline([("classifier", cls)])
for i in range(train_size):
    pipline.partial_fit(trains[2* i], trains[2*i+1], classes=([0,1]))
joblib.dump(pipline, joblib_path)


here is my code ,i can get the right joblib file ,while it cannot get any pmml file
should i use cls.partial_fit(trains[2* i], trains[2*i+1], classes=([0,1])) in line 3 ?

**Comment by vruusmann** [comment]
Doesn't work (invoking `PMMLPipeline.fit`):
```python
cls = SGDClassifier(..)
pipeline = PMMLPipeline([("classifier", cls)])
pipeline.fit(X, y)

sklearn2pmml(pipeline, ..)
```

Works (invoking `SGDClassifier.fit`):
```python
cls = SGDClassifier(..)
cls.partial_fit(X, y)
pipeline = PMMLPipeline([("classifier", cls)])

sklearn2pmml(pipeline, ..)
```

> That the PMMLPipeline class should provide a partial_fit(X, y) method (in addition to the fit(X, y) method)?

Checked the SkLearn 0.19.2 documentation - class `sklearn.pipeline.Pipeline` does not specify a `partial_fit` method, which means that class `sklearn2pmml.pipeline.PMMLPipeline` shouldn't be providing it neither.

Closing this issue as solved.
